src/core/service.c

```diff
@@ -2644,6 +2644,10 @@ static void service_sigchld_event(Unit *u, pid_t pid, int code, int status) {
         else
                 assert_not_reached("Unknown code");
 
+        if ((f != SERVICE_SUCCESS) && UNIT(s)->fragment_path &&
+                streq(basename(UNIT(s)->fragment_path), SPECIAL_INITRD_SWITCH_ROOT_SERVICE))
+                f = SERVICE_SUCCESS;
```

Comment:
So the goal here is to ignore the return code from initrd-switch-root.service? That seems very iffy. But I don't remember this old code base too well. Maybe @msekletar  should have a look?